feat : 댓글 Create 기능 개발

jade-coding opened the pull request:
## PR Summary

- 댓글을 Create하여 DB에 저장하는 기능을 개발했습니다. 
- resolves #1 

## Changes

- Comment 엔티티 생성
- Comment 리포지토리 생성
- Spring Data Jpa가 제공하는 기능 사용
- Comment 서비스 생성
- 댓글 저장 기능 구현 및 테스트
- Post <-> Comment 일대다 맵핑 

## Test Checklist
- [x] 댓글 저장 성공 테스트

jade-coding committed:
feat : 댓글 엔티티 생성

jade-coding committed:
feat : 댓글 리포지토리 생성

jade-coding committed:
feat : 댓글 저장 기능 구현

jade-coding committed:
test : 댓글 저장 기능 테스트

an2z commented on back/blog/src/main/java/saramdle/blog/domain/Comment.java:
```diff
@@ -0,0 +1,44 @@
+    private LocalDateTime createAt;
+
+    private LocalDateTime updateAt;
+
```
Comment 스키마와 필드가 안맞는거 같아요!
<img width="790" alt="image" src="https://user-images.githubusercontent.com/91456818/226260518-abf7dd04-9a7a-41dc-998e-2c001c50aedf.png">

jade-coding replied:
commentId -> id로 하기로 했던 것 같은데 문서에도 수정해놓을게요

an2z commented on back/blog/src/test/java/saramdle/blog/service/CommentServiceTest.java:
```diff
@@ -0,0 +1,25 @@
+        assertThat(saveId).isEqualTo(comment.getId());
+    }
+
+}
```
마지막 개행을 추가해주세요!

an2z reviewed (changes requested):
리뷰 확인 부탁드릴게요.😄

jade-coding committed:
feat : Post, Comment 일대다 연관관계 맵핑  

- Comment 엔티티에 Post 엔티티와 맵핑
- 생성자 호출시 Post가 주입

jade-coding committed:
chore : Comment 엔티티 필드명 수정

- createAt -> createdAt 변경
- updateAt -> updatedAt 변경

jade-coding committed:
test : Comment Create 기능 테스트 수정

- 변경 전) comment post 맵핑 없이 DB에 저장  
- 변경 후) comment - post 맵핑 후 comment에 post가 주입되어져서 DB에 저장

an2z commented on back/blog/src/test/java/saramdle/blog/service/CommentServiceTest.java:
```diff
@@ -1,25 +1,45 @@
+    PostService postService;
+
+
+    @BeforeEach
+    void beforeEach(){
+        Post post1 = Post.builder().build();
+        Long savedId1 = postService.save(post1);
+        Assertions.assertThat(post1.getId()).isEqualTo(savedId1);
+    }
+
```
Post 저장에 대한 테스트는 CommentServiceTest의 역할이 아니라고 생각하는데, 상우님은 어떻게 생각하시나요?!

an2z commented on back/blog/src/test/java/saramdle/blog/service/CommentServiceTest.java:
```diff
@@ -1,25 +1,45 @@
+    @Autowired
+    PostService postService;
+
+
```
한줄이 아닌 두줄을 띄워주셨는데, 따로 이유가 있으신걸까요?

jade-coding committed:
chore : Comment 엔티티 필드 수정 

- title, imgUrl, author 필드 삭제
- contents -> content 필드명 변경
- commentWriter 추가

an2z reviewed (commented):
작성해주신 코드에 대해 저의 의견을 남겼습니다!
제 의견이 정답은 아니니 코드 작성의 본인만의 이유를 설명해주시면 좋을거 같아요.
